pkg/kubelet/dockertools/manager.go

```diff
@@ -1456,14 +1455,13 @@ func (dm *DockerManager) pullImage(pod *api.Pod, container *api.Container) error
 		return nil
 	}
 
-	// TODO get the pull secrets from the container's ImageSpec and the pod's service account
-	err = dm.PullImage(spec, nil)
+	err = dm.PullImage(spec, pullSecrets)
 	dm.runtimeHooks.ReportImagePull(pod, container, err)
 	return err
 }
 
 // Sync the running pod to match the specified desired pod.
-func (dm *DockerManager) SyncPod(pod *api.Pod, runningPod kubecontainer.Pod, podStatus api.PodStatus) error {
+func (dm *DockerManager) SyncPod(pod *api.Pod, runningPod kubecontainer.Pod, podStatus api.PodStatus, pullSecrets []api.Secret) error {
```

Comment:
It feels weird to me that SyncPod gets the secrets passed in. Is there a reason why SyncPod can't get the secrets itself? Or better yet, can that be pushed all the way down into the pullImage call?